### PR TITLE
Tweaks the expressroutes validateInput behavior to correctly process …

### PR DESCRIPTION
**Pull request by l12s**
…values in the query string.

Closes krakenjs/swaggerize-express/issues/64

### CODE REVIEW

**Comment by tlivings** [comment]
Building failing :(

Also at this point, not much need for `isPath`, `isForm`, etc... The purpose of these was when multiple `parameter.in` types were being lumped together into a single type. At this point they are entirely distinct.

**Comment by l12s** [comment]
My bad, forgot to run tests locally first. I've tweaked the test to reflect the value coercion on req.query that's now happening in expressroutes.

I'm not quite following re. the isForm, isPath piece - were you thinking of just swapping in another switch to pass through the coerced values from the validator there?

**Comment by l12s** [comment]
See above for what I'd suggest re dropping isPath etc. Basically moves to an accessor pattern, drops the switch logic into an object lookup and breaks up the longer function into smaller fn calls.

LMK if it seems reasonable, also happy to tweak to fit another preferred pattern.

**Comment by l12s** [comment]
hmm, will look at the travis errors. tests seem happy locally

**Comment by l12s** [comment]
couple of lint issues, now fixed

**Comment by tlivings** [comment]
I like this, but can you remove `lodash` usage? It was an unused dependency left over that needed to be removed.

**Comment by tlivings** [comment]
Also, sorry this is taking so long! I appreciate the patience.

**Comment by l12s** [comment]
no problem. I've pushed a version w/out lodash now - I left the dep in the package.json file for now though, lmk if you'd like me to remove.

If this seems good I'll squash up the commits and get it ready for merge.

File: lib/expressroutes.js [hunk 1]
```diff
@@ -1,9 +1,116 @@
...
-    async = require('async');
+    utils = require('swaggerize-routes/lib/utils');
+
+
+/**
+ * Makes default accessor functions for a specific data location, e.g. query, params etc
+ * @param dataLoc
+ * @returns {{get: get, set: set}}
+ */
+var defaultAccessor = function(dataLoc) {
```
Comment (tlivings) [inline lib/expressroutes.js]: Prefer these as simply named functions (as per rest of code style).

**Comment by tlivings** [comment]
On the face of it, seems ok (with exception of code style comment). Will get another from team to review as well!

**Comment by grawk** [comment]
LGTM. Code consistency is important but I would argue for functions assigned to variables.. Maybe merge this and we can argue about that later.. :+1:

**Pull request merged**